### DISCUSSION TITLE
Apple 12 Pro can't discover HomeSpen device.

### DISCUSSION THREAD

**Discussion by s887432** [HomeSpan/HomeSpan#692] [Q&A]
Hi Guys,

I am using ESP32 Wroom 32 for HomeSpen development.
Tried to download 01-SimpleLightBulb example to device.
Unfortunately, my phone can't find any device.
tried to erase all before download.
tried to use new boards.
same issue happened.

it might relative to my phone.
it's IOS 17.1.1.

Is there any one have similar experience?
any suggestion or comment?

Thank you very much.

Patrick

**Comment by HomeSpan** [comment]
I've not had any problems with iOS 17.1.1.  Here are some steps you can take:

1. Check that HomeSpan is properly running by typing 'i', 's', or 'd' into the Serial Monitor (or even just '?' to get a list of the Serial Monitor commands)
2. Make sure your device is not already paired -- if it is, type 'U' or 'H' from the Serial Monitor to erase all pairing info.
3. Make sure nothing is connected to your HomeSpan device (except of course the USB cable).  In other words, no external LEDs, switches, etc.
4. Confirm you've been able to connect HomeSpan to your WiFi network.  To double-check, you may want to enable the Web Log option (see the API for details) so you can ensure connectivity to HomeSpan by using your browser to access the HomeSpan Web Log
5. Make sure your iPhone is running on the same network as the ESP32.  If your WiFi supports both 2.4 and 5 GHz bands, you must ensure they can cross talk, else the iPhone will likely not be able to find the ESP32, which only uses the 2.4 GHz band
6. If you have a mesh network, make sure it is also configured to allow full cross-talk between all bands and access points
7. If everything looks good from WiFi, next is to ensure MDNS records from the ESP32 are being properly broadcast to all other devices.  There is a free app for the iPhone called Discovery DNS-SD Browser.  Download this program and use it to check the `_hap._tcp` service list.  You should see MDNS records from HomeSpan on your iPhone.

Problems with MDNS are unfortunately not uncommon.  MDNS records must be able to be transmitted properly across your WiFi network for the iPhone to find HomeSpan.  Many routers do not properly transmit these records, or they include default filters that block the records for full transmission.  A search of the web for your particular router may help inform on whether it is compatible with WiFi based HomeKit devices.  Some routers just won't work with HomeKit at all.  Others need to be properly configured to avoid issues, etc.  And sometimes it is very specific to a particular installation.  There is a lot about this on the web, but I have not found a central source of info or ways of addressing this issue in all circumstances.  If you look through the Issues list for HomeSpan you'll find that sometimes users bought a different router and solved all their HomeKit problems.

**Reply by s887432** [reply]
Hi,

here are the result from your receommanded.

**Command i:**
*** HomeSpan Info ***

➤ Accessory:  AID=1
   ➟ Service AccessoryInformation:  IID=1, UUID="3E"
      ⇨ Characteristic Identify(0):  IID=2, UUID="14", Perms=PW
   ➟ Service LightBulb:  IID=3, UUID="43"
      ⇨ Characteristic On(0):  IID=4, UUID="25", Perms=PR+PW+EV
➤ Accessory:  AID=2
   ➟ Service AccessoryInformation:  IID=1, UUID="3E"
      ⇨ Characteristic Identify(0):  IID=2, UUID="14", Perms=PW
   ➟ Service Switch:  IID=3, UUID="49"
      ⇨ Characteristic On(0):  IID=4, UUID="25", Perms=PR+PW+EV

Configured as Bridge: NO
Configuration Number: 1

Database Validation:  Warnings=0, Errors=0

Service                             UUID         AID  IID  Update  Loop  Button  Linked Services
------------------------------  --------  ----------  ---  ------  ----  ------  ---------------
AccessoryInformation                  3E           1    1      NO    NO      NO  -
LightBulb                             43           1    3     YES    NO      NO  -
AccessoryInformation                  3E           2    1      NO    NO      NO  -
Switch                                49           2    3     YES    NO      NO  -

*** End Info ***

**Command s:**

*** HomeSpan Status ***

IP Address:        0.0.0.0

Accessory ID:      16:6B:26:D0:CE:E1                               LTPK: 2C14558514AB333ACEB9E49121F46B4114DA6C21433CBB5208CCB92C08C192AF
No Paired Controllers

Connection #0 (unconnected)
Connection #1 (unconnected)
Connection #2 (unconnected)
Connection #3 (unconnected)
Connection #4 (unconnected)
Connection #5 (unconnected)
Connection #6 (unconnected)
Connection #7 (unconnected)
Connection #8 (unconnected)
Connection #9 (unconnected)
Connection #10 (unconnected)
Connection #11 (unconnected)
Connection #12 (unconnected)
Connection #13 (unconnected)

*** End Status ***

**command d:**
*** Attributes Database: size=483  configuration=1 ***

{
  "accessories":[
    {
      "aid":1,
      "services":[
        {
          "iid":1,
          "type":"3E",
          "characteristics":[
            {
              "iid":2,
              "type":"14",
              "format":"bool",
              "perms":[
                "pw"
              ]
            }
          ]
        },
        {
          "iid":3,
          "type":"43",
          "characteristics":[
            {
              "iid":4,
              "type":"25",
              "value":0,
              "format":"bool",
              "perms":[
                "pr",
                "pw",
                "ev"
              ]

**command U:**

*** HomeSpan Pairing Data DELETED ***


DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP

**command H:**


*** HomeSpan Device ID and Pairing Data DELETED!  Restarting...

ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13260
load:0x40080400,len:3028
entry 0x400805e4

************************************************************
Welcome to HomeSpan!
Apple HomeKit for the Espressif ESP-32 WROOM and Arduino IDE
************************************************************

** Please ensure serial monitor is set to transmit <newlines>

Message Logs:     Level 0
Status LED:       Pin -  *** WARNING: Status LED Pin is UNDEFINED
Device Control:   Pin -  *** WARNING: Device Control Pin is UNDEFINED
Sketch Version:   n/a
HomeSpan Version: 1.8.0
Arduino-ESP Ver.: 2.0.14
ESP-IDF Version:  4.4.6
ESP32 Chip:       ESP32-D0WDQ6 Rev 1 dual-core 4MB Flash
ESP32 Board:      doitESP32devkitV1
PWM Resources:    16 channels, 8 timers, max 20-bit duty resolution
Sodium Version:   1.0.12-idf  Lib 9.4
MbedTLS Version:  mbed TLS 2.28.4
Sketch Compiled:  Nov 14 2023 21:36:45
Partition:        app0

**Reply by s887432** [reply]
only connect USB cable to laptop only. no extra components or wire connected.

does the WiFi connection link after phone pair success?
can't find device in first page, the issue should be not relative to WiFi?

BTW, my phone connect to 2.4G WiFi.
no mesh used

i can get something from Discovery DNS-SD Browser

HomePodSensor 165032
._hap._tcp.local. 
sanlou.local:54779 
192.168.50.145:54779 
fe80::14a6:b12b:6ac1:c3a7%en0:54779 
fd18:49ad:1a0f:4cff:2:fd85:1151:a8a:54779 
c# = 1 
ci = 10 
ff = 2 
id = D3:B5:B9:44:9F:3D 
md = HomePod 
pv = 1.1 
s# = 3 
sf = 0 
sh = NnEtWA==

**Comment by HomeSpan** [comment]
Based on the log file above, you are not connected to WiFi - it's showing an address of 0.0.0.0.

**Reply by s887432** [reply]
Dear Sir,

yes. the issue has been fixed after assigned corrent WiFi configurateion.
thank you very much for your support.

best regards,

Patrick
